### PR TITLE
fix: add missing package.json and source maps

### PR DESCRIPTION
**Pull request by code-asher**
These issues were caused when I switched Code's packaged build instead of putting it together ourselves.

See individual commits for more information.

### CODE REVIEW

File: ci/build/build-release.sh [hunk 1]
```diff
@@ -78,11 +78,25 @@ EOF
...
   mkdir -p "$VSCODE_OUT_PATH"
 
-  # - Some extensions have a .gitignore which excludes their built source from
-  #   the npm package so exclude any .gitignore files.
-  # - Exclude Node as we will add it ourselves for the standalone and will not
-  #   need it for the npm package.
-  rsync -avh --exclude .gitignore --exclude /node ./lib/vscode-reh-web-*/ "$VSCODE_OUT_PATH"
+  local rsync_opts=()
+  if [[ ${DEBUG-} = 1 ]]; then
+    rsync_opts+=(-vh)
```
[the remark this answers is not in the document]
Comment (jsjoeio) [reply ci/build/build-release.sh:83]: Ah:
- `-h`: -h, --human-readable        output numbers in a human-readable format
- `-v`: -v, --verbose               increase verbosity

Source: https://linux.die.net/man/1/rsync

File: ci/build/build-release.sh [hunk 2]
```diff
@@ -78,11 +78,25 @@ EOF
...
+  # Exclude Node as we will add it ourselves for the standalone and will not
+  # need it for the npm package.
+  rsync_opts+=(--exclude /node)
+
+  # Exclude Node modules.
+  if [[ $KEEP_MODULES = 0 ]]; then
+    rsync_opts+=(--exclude node_modules)
+  fi
+
+  rsync "${rsync_opts[@]}" ./lib/vscode-reh-web-*/ "$VSCODE_OUT_PATH"
```
Comment (jsjoeio) [inline ci/build/build-release.sh:99]: This approach looks a lot cleaner 👏🏼

File: ci/build/npm-postinstall.sh [hunk 3]
```diff
@@ -98,14 +98,6 @@ vscode_yarn() {
...
   cd extensions
   yarn --production --frozen-lockfile
-
-  for ext in */; do
-    ext="${ext%/}"
-    echo "extensions/$ext: installing dependencies"
-    cd "$ext"
-    yarn --production --frozen-lockfile
-    cd "$OLDPWD"
-  done
```
Comment (jsjoeio) [inline ci/build/npm-postinstall.sh:108]: beautiful find

File: patches/display-language.diff [hunk 4]
```diff
@@ -79,7 +79,7 @@ Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.html
 +		} catch (error) { /* Probably fine. */ }
  		Object.keys(self.webPackagePaths).map(function (key, index) {
  			self.webPackagePaths[key] = new URL(
- 				`{{VS_BASE}}/static/remote/web/node_modules/${key}/${self.webPackagePaths[key]}`,
+ 				`{{VS_BASE}}/static/node_modules/${key}/${self.webPackagePaths[key]}`,
```
Comment (jsjoeio) [inline patches/display-language.diff:82]: Does this need to be changed in my upstream PR?

File: ci/build/build-release.sh [hunk 5]
```diff
@@ -145,6 +145,11 @@ EOF
 
   rsync "$VSCODE_SRC_PATH/remote/yarn.lock" "$VSCODE_OUT_PATH/yarn.lock"
 
+  # Include global extension dependencies as well.
+  rsync "$VSCODE_SRC_PATH/extensions/package.json" "$VSCODE_OUT_PATH/extensions/package.json"
+  rsync "$VSCODE_SRC_PATH/extensions/yarn.lock" "$VSCODE_OUT_PATH/extensions/yarn.lock"
+  rsync "$VSCODE_SRC_PATH/extensions/postinstall.js" "$VSCODE_OUT_PATH/extensions/postinstall.js"
+
```
Comment (jsjoeio) [inline ci/build/build-release.sh:150]: Nice catch 👏🏼

File: ci/build/build-release.sh [hunk 6]
```diff
@@ -136,10 +136,8 @@ EOF
   ) > "$VSCODE_OUT_PATH/product.json"
 
   # Use the package.json for the web/remote server.  It does not have the right
-  # version though so pull that from the main package.json.  Also remove keytar
-  # since the web does not rely on it and that removes the dependency on
-  # libsecret.
-  jq --slurp '.[0] * {version: .[1].version} | del(.dependencies.keytar)' \
+  # version though so pull that from the main package.json.
+  jq --slurp '.[0] * {version: .[1].version}' \
```
Comment (jsjoeio) [inline ci/build/build-release.sh:140]: Nice catch! How'd you figure this out? That log message when code-server starts up? IIRC

**Review by jsjoeio — APPROVED** [review]
Nothing concerning, just a bunch of questions. Nice work!

File: patches/display-language.diff (hunk 4 above)
Comment (code-asher) [reply patches/display-language.diff:82]: Nope, this is not part of the changes this patch applies, it is just part of the surrounding context.  So this change is from an earlier patch.

File: ci/build/build-release.sh (hunk 6 above)
Comment (code-asher) [reply ci/build/build-release.sh:140]: Fortunately we still have libsecret as a dependency in our docs (as far as I could tell, hopefully I did not miss a spot) so nothing else changes

File: ci/build/npm-postinstall.sh (hunk 3 above)
Comment (jsjoeio) [reply ci/build/npm-postinstall.sh:108]: Good idea. I'll add a note to the release issue

**Pull request merged**